Compiling outside of a git repo fails

Because of the `RIOT_VERSION` variable not available, the compilation outside of a git repo (for instance from the zip file downloaded fron github) fails.

```
Building application "hello-world" for "native" with MCU "native".

/bin/sh: 4: Syntax error: "(" unexpected
/home/vincent/Documents/RIOT-master/Makefile.base:64 : la recette pour la cible « /home/vincent/Documents/RIOT-master/examples/hello-world/bin/native/hello-world/main.o » a échouée
make[1]: *** [/home/vincent/Documents/RIOT-master/examples/hello-world/bin/native/hello-world/main.o] Erreur 2
/home/vincent/Documents/RIOT-master/examples/hello-world/../../Makefile.include:262 : la recette pour la cible « all » a échouée
make: *** [all] Erreur 2
```

I found that using the `RIOT_VERSION_OVERRIDE` fixes the issue, but the message is not clear. Is it known / a feature ?

--- a/Makefile.include
+++ b/Makefile.include
@@ -510,13 +510,14 @@ $(RIOTBUILD_CONFIG_HEADER_C):
 	@mkdir -p '$(dir $@)'
 	$(AD)'$(RIOTBASE)/dist/tools/genconfigheader/genconfigheader.sh' '$@' $(CFLAGS_WITH_MACROS)
 
+CFLAGS_WITH_MACROS := $(CFLAGS)
+
 ifneq (,$(RIOT_VERSION_OVERRIDE))
-export CFLAGS += -DRIOT_VERSION=\"$(RIOT_VERSION_OVERRIDE)\"
+export CFLAGS_WITH_MACROS += -DRIOT_VERSION=\"$(RIOT_VERSION_OVERRIDE)\"
 else
-export CFLAGS += -DRIOT_VERSION=\"$(RIOT_VERSION)\"
+export CFLAGS_WITH_MACROS += -DRIOT_VERSION=\"$(RIOT_VERSION)\"
 endif
 
-CFLAGS_WITH_MACROS := $(CFLAGS)
 CFLAGS := $(patsubst -D%,,$(CFLAGS))
 CFLAGS := $(patsubst -U%,,$(CFLAGS))
 CFLAGS += -include '$(RIOTBUILD_CONFIG_HEADER_C)'

